c/driver/postgresql/database.h

```diff
@@ -58,12 +62,29 @@ class PostgresDatabase {
     return type_resolver_;
   }
 
-  AdbcStatusCode RebuildTypeResolver(struct AdbcError* error);
+  Status InitVersions(PGconn* conn);
+  Status RebuildTypeResolver(PGconn* conn);
+  std::string_view VendorName() {
+    if (redshift_server_version_[0] != 0) {
+      return "Redshift";
+    } else {
+      return "PostgreSQL";
+    }
+  }
+  const std::array<int, 3>& VendorVersion() {
+    if (redshift_server_version_[0] != 0) {
+      return redshift_server_version_;
+    } else {
+      return postgres_server_version_;
+    }
+  }
 
  private:
   int32_t open_connections_;
   std::string uri_;
   std::shared_ptr<PostgresTypeResolver> type_resolver_;
+  std::array<int, 3> postgres_server_version_{};
+  std::array<int, 3> redshift_server_version_{};
```

Comment:
I suppose if you wanted to be fancy you could use std::variant with single-member structs but this works fine